Update instructions for making a release (#813)
ESMValGroup/ESMValCore: doc/contributing.rst

To make a new release of the package, follow these steps:

1. Check the tests on GitHub Actions and CircleCI
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Check the ``nightly``
`build on CircleCI <https://circleci.com/gh/ESMValGroup/ESMValCore/tree/master>`__
and the
`GitHub Actions run <https://github.com/ESMValGroup/ESMValCore/actions>`__.
All tests should pass before making a release (branch).

2. Increase the version number
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The version number is stored in ``esmvalcore/_version.py``,
``package/meta.yaml``, ``CITATION.cff``. Make sure to update all files.
Also update the release date in ``CITATION.cff``.
See https://semver.org for more information on choosing a version number.
Make a pull request and get it merged into ``master``.

3. Add release notes
~~~~~~~~~~~~~~~~~~~~
Use the script
`esmvaltool/utils/draft_release_notes.py <https://docs.esmvaltool.org/en/latest/utils.html#draft-release-notes-py>`__

> to create create a draft of the release notes.

Reviewer: probably best to add that we need `pygithub` and can be installed by running `pip install pygithub` and a personal token obtainable by following instructions at https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line (I know the script is verbose and tells you that, but let's have the release manager have the tools installed already and not see it fail the first time)

Author: I've improved the documentation here: https://github.com/ESMValGroup/ESMValTool/pull/1848

Reviewer: yah that's nice, just took a look, good one, man!

This script uses the titles and labels of merged pull requests since the
previous release.
Review the results, and if anything needs changing, change it on GitHub and
re-run the script until the changelog looks acceptable.
Copy the result to the file ``doc/changelog.rst``.

> Make a pull request and get it merged into ``master``.

Reviewer: worth saying they should add at least two of the Core Team as PR reviewers

Author: I should hope that someone making the release knows how to get something merged. However, maybe we should always have the release manager of the previous release as a 'backup release manager', doing it toghether like we are for this release. That would ensure some continuity.

Author: OK, I added this

3. Create a release branch
~~~~~~~~~~~~~~~~~~~~~~~~~~
Create a branch off the ``master`` branch and push it to GitHub.
Ask someone with administrative permissions to set up branch protection

> rules for it so only you and one other person can push to it.

Reviewer: ```suggestion
rules for it so only you and the admin can push to it.
```

Author: That would be a bit vague, who is 'the admin'? shall I write, 'the release manager of the previous release' or 'the backup release manager' instead?

Reviewer: yeps, sorry, admin is supervague yeah

Author: Added!
